### PR TITLE
Fix two bugs in column expr predicates

### PR DESCRIPTION
**Pull request by dirtysalt**
This PR fixes two bugs. And these two bugs are found when we are running our daily tests.

1. Don't early return if the size of chunk filtered by the expr predicates is 0. Maybe there are more data in the segment files.
2. Dict values from dict columns are non-padded. So when we rewrite predicates on dict column, we should use non-padded string, instead of zero-padded string.

### CODE REVIEW

File: be/src/storage/rowset/vectorized/segment_iterator.cpp [hunk 1]
```diff
@@ -820,40 +820,43 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
...
+
+    size_t chunk_size = _filter_by_expr_predicates(chunk, rowid);
 
     _opts.stats->block_load_ns += sw.elapsed_time();
 
     int64_t total_read = _opts.stats->raw_rows_read - prev_raw_read;
 
-    DCHECK_EQ(chunk_start, chunk->num_rows());
-    if (UNLIKELY(chunk_start == 0)) {
+    if (UNLIKELY(raw_chunk_size == 0)) {
```
Comment (kangkaisen) [inline be/src/storage/rowset/vectorized/segment_iterator.cpp:832]: I think we could move this check to line 823, then we don't need the raw_chunk_size variate?

File: be/src/storage/vectorized/column_predicate_rewriter.cpp [hunk 2]
```diff
@@ -160,11 +161,12 @@ bool ColumnPredicateRewriter::_rewrite_predicate(ObjectPool* pool, const FieldPt
         }
         if (PredicateType::kLE == pred->type() || PredicateType::kLT == pred->type()) {
             _get_segment_dict(&sorted_dicts, _column_iterators[cid]);
-            std::vector<std::string> str_codewords;
-            auto value = pred->value().get_slice().to_string();
+            // use non-padding string value.
+            auto value = pred->values()[0].get_slice().to_string();
```
Comment (kangkaisen) [inline be/src/storage/vectorized/column_predicate_rewriter.cpp:165]: I think we should add some comments to `value()` and `values()` method in `ColumnPredicate`, let other people know the char type has the padding issue,  the `value()` and `values()`  are different

File: be/src/storage/rowset/vectorized/segment_iterator.cpp (hunk 1 above)
Comment (dirtysalt) [reply be/src/storage/rowset/vectorized/segment_iterator.cpp:832]: It's okay. But then `block_load_ns` will not be updated.  I was thinking it's better to make metrics as accurate as possible.

```
    _opts.stats->block_load_ns += sw.elapsed_time();
```

File: be/src/storage/vectorized/column_predicate_rewriter.cpp (hunk 2 above)
Comment (dirtysalt) [reply be/src/storage/vectorized/column_predicate_rewriter.cpp:165]: OK.

**Comment by kangkaisen** [comment]
run  starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Comment by kangkaisen** [comment]
run starrocks_fe_unittest

**Review by trueeyu — APPROVED** [review]

**Pull request merged**